server/webhook.go

```diff
@@ -63,6 +63,12 @@ func (s *Server) registerWebhookRoutes(g *echo.Group) {
 			return echo.NewHTTPError(http.StatusNotFound, fmt.Sprintf("Webhook endpoint not found: %v", webhookEndpointID))
 		}
 
+		if branch, err := parseBranchNameFromRefs(pushEvent.Ref); err != nil {
+			return echo.NewHTTPError(http.StatusBadRequest, "Invalid ref").SetInternal(err)
+		} else if branch != repo.BranchFilter {
+			return c.String(http.StatusOK, "")
```

Comment:
I thought the GitLab branch filter supports more complex filtering syntax than equivalence, https://stackoverflow.com/a/61309858 . Generally, we need to take care of all of them, but if we can confirm with existing users that everyone only uses exact matches then this line of change is fine.